post.md

```diff
@@ -0,0 +1,427 @@
+# Server Actions with User Confirmations in Next.js 13
+
+Next.js 13 has brought exciting new experimental React features to web development. The combination of server side rendered components and server actions have the potential to simplify web development while still giving us the full power of client side React when we need it.
+
+This article introduces the basics of server actions, and how to use them in combination with actions that require user confirmation. This is a common scenario for destructive actions such as deleting a record from a database.
+
+## Sample application
+
+This article presumes you're acquainted with setting up and running Next.js applications.
+
+To view the code referenced in this article, [checkout the repo](https://github.com/CodingZeal/nextjs-example-confirm-server-action).
+
+This article and accompanying example application utilize Next.js version 13.5.4. Bear in mind that server actions remain in the experimental phase and could evolve.
+
+## A Glimpse at Server Actions
+
+Next.js's server actions empower developers to execute server-side operations directly from the client without the hassle of wiring up HTTP client code. As the Next.js documentation succinctly puts it:
+
+> With Server Actions, you don't need to manually create API endpoints. Instead, you define asynchronous server functions that can be called directly from your components.
+
+### Required Next.js Configurations
+
+To leverage server actions, the experimental feature flag must be enabled in your Next.js configuration:
+
+```ts
+module.exports = {
+  experimental: {
+    serverActions: true,
+  },
+};
+```
+
+### Basic Server Action Example
+
+Consider the following rudimentary example that sends form data from the client to the server action and then renders to the server console. Note, the server action is set as the form's `action`:
+
+```tsx
+export default function SomePage() {
+  async function exampleServerFunction(formData: FormData) {
+    "use server";
+
+    console.log("formData", Array.from(formData));
+  }
+
+  return (
+    <div>
+      <form action={exampleServerFunction}>
+        <div>
+          <label htmlFor="name">Name: </label>
+          <input type="text" name="name" />
+        </div>
+
+        <button type="submit">Submit</button>
+      </form>
+    </div>
+  );
+}
+```
+
+## Using the experimental `useFormStatus` hook for user feedback
+
+At its core, `useFormStatus` tells us the current state of our form's server action, allowing for dynamic UI updates based on server action outcomes.
+
+The hook currently returns a `pending` property, which is set to true during the form submission process. This can be incredibly useful for preventing multiple submissions and giving users feedback that the form submission is still in process.
+
+Before we continue here is the server action we are calling:
+
+```ts
+"use server";
```

Comment:
It may be helpful to drop a brief note about what the `'use server'` directive is doing for this component.

Reply:
Going to leave this off and let the reader get it from Next.js docs. Tried to integrate it into the content but didn't flow as well.